src/coreclr/vm/jitinterface.cpp

```diff
@@ -8241,17 +8241,30 @@ void CEEInfo::reportInliningDecision (CORINFO_METHOD_HANDLE inlinerHnd,
 
         if (CORProfilerEnableRejit())
         {
-            // If ReJIT is enabled, there is a chance that a race happened where the profiler
-            // requested a ReJIT on a method, but before the ReJIT occurred an inlining happened.
-            // If we end up reporting an inlining on a method with non-default IL it means the race
-            // happened and we need to manually request ReJIT for it since it was missed.
-            CodeVersionManager* pCodeVersionManager = pCallee->GetCodeVersionManager();
-            CodeVersionManager::LockHolder codeVersioningLockHolder;
-            ILCodeVersion ilVersion = pCodeVersionManager->GetActiveILCodeVersion(pCallee);
-            if (ilVersion.GetRejitState() != ILCodeVersion::kStateActive || !ilVersion.HasDefaultIL())
+            ModuleID modId = 0;
+            mdMethodDef methodDef = mdMethodDefNil;
+            BOOL shouldCallReJIT = FALSE;
+
+            {
```

Comment:
Would it make sense to leave a note that we are releasing the CodeVersionManager lock (via exiting scope) before acquiring the ReJIT lock to protect against a deadlock, in case some future refactoring might overlook why `ReJitManager::RequestReJIT` is separated?

Reply:
Alternatively, an assert would be enough to document that and validate callers don't try to grab the lock either.

Reply:
Added a comment